lib/streamlit/runtime/caching/cache_utils.py

```diff
@@ -349,7 +470,10 @@ def maybe_show_cached_st_function_warning(
             The name of the Streamlit function that was called.
 
         """
-        if st_func_name in NONWIDGET_ELEMENTS:
+        # There are some elements not in either list, which we still want to warn about.
+        # Ideally we will fix this by either updating the lists or creating a better
+        # way of categorizing elements.
+        if st_func_name in NONWIDGET_ELEMENTS or st_func_name in WIDGETS:
```

Comment:
Just noticed that I still need to update this to only skip warning about widgets if they are allowed